### DISCUSSION TITLE
Classic Titanium App to Alloy

### DISCUSSION THREAD

**Discussion by martinrademacher** [appcelerator/titanium_mobile#12730] [Q&A] [answered]
Hi,

is there an in detail documentation how to perform an update from Classic Titanium App to Alloy?
I already know https://devblog.axway.com/mobile-apps/moving-your-classic-titanium-app-to-alloy/ but especially Step 1 seems to lack some crucial information.

thank you in advance

**Comment by m1ga** [comment, marked as answer]
it would help a bit if you tell us where you are stuck. Haven't used a classic setup in ages but what I do is:
* create a new Alloy app (`appc new --no-services`)
* copy all your old js files to `app/lib` (like in step 1) so basically all the files you `require` in your code
* copy the code from `app.js` into `index.js` since that is the new starting point of your Alloy app
* remove everything besides `<alloy />` from `views/index.xml`

ignore all the views and styles for now, that is the stuff that follows in step 2 of the tutorial. Oh and move your assets. Depending on your code and the SDK you were using it should already build and run now.

After that it is time to move all the `Ti.UI.createWindow()` stuff into `<Window>` and the properties to the `index.tss` and so on. This will be the major work and the tutorial covers those steps, too.

**Reply by martinrademacher** [reply]
thank you. starting with a new Alloy App was the first step which was missing in the documentation ;) I'm defining a lot of globals in my app.js so I moved all my app.js to alloy.js and changed global. to Alloy.Globals. but now it seems to work. 
thanks

**Reply by m1ga** [reply]
Nice! If you get used the the structure/syntax it will be so much more fun to maintain the app! Especially if new people look at the code
